### PR TITLE
Fix bad tx behavior through tx pool cleaning

### PR DESCRIPTION
**Pull request by SebastianMarian**
* Refactored some misleading methods name compared with their behavior in TransactionCoordinator and PreProcess interface/implementations
* Fixed bad tx behavior through tx pool cleaning for each processed tx at commit time

**How to be tested:**

We should start an #allin system test with an additional log level set to: preprocess:TRACE
1) After a while we should verify the tx pools and they should remain almost empty. 
2) Another check should be done if already processed txs in previous block, would not be delivered as good to be processed in the next block to the next proposer by the SortedTransactionsProvider component. **(grep after "bad tx")**

### CODE REVIEW

**Review by sasurobert — APPROVED** [review]

**Review by bogdan-rosianu — CHANGES_REQUESTED** [review]

File: process/block/baseProcess.go [hunk 1]
```diff
@@ -732,6 +732,20 @@ func (bp *baseProcessor) removeBlockDataFromPools(headerHandler data.HeaderHandl
 	return nil
 }
 
+func (bp *baseProcessor) removeTxsFromPools(bodyHandler data.BodyHandler) error {
+	body, ok := bodyHandler.(*block.Body)
+	if !ok {
+		return process.ErrWrongTypeAssertion
+	}
+
+	err := bp.txCoordinator.RemoveTxsFromPool(body)
```
[the remark this answers is not in the document]
Comment (SebastianMarian) [reply process/block/baseProcess.go:741]: Done

File: process/block/preprocess/basePreProcess.go [hunk 2]
```diff
@@ -45,19 +45,35 @@ type basePreProcess struct {
 	pubkeyConverter      core.PubkeyConverter
 }
 
-func (bpp *basePreProcess) removeDataFromPools(
+func (bpp *basePreProcess) removeBlockDataFromPools(
 	body *block.Body,
 	miniBlockPool storage.Cacher,
 	txPool dataRetriever.ShardedDataCacherNotifier,
 	isMiniBlockCorrect func(block.Type) bool,
 ) error {
```
[the remark this answers is not in the document]
Comment (SebastianMarian) [reply process/block/preprocess/basePreProcess.go:54]: Done

File: process/block/preprocess/basePreProcess.go [hunk 3]
```diff
@@ -70,6 +86,29 @@ func (bpp *basePreProcess) removeDataFromPools(
...
+
+	return nil
+}
+
+func (bpp *basePreProcess) removeMiniBlocksFromPools(
+	body *block.Body,
+	miniBlockPool storage.Cacher,
+	isMiniBlockCorrect func(block.Type) bool,
+) error {
+
```
[the remark this answers is not in the document]
Comment (SebastianMarian) [reply process/block/preprocess/basePreProcess.go:99]: Done

File: process/block/preprocess/basePreProcess.go [hunk 4]
```diff
@@ -70,6 +86,29 @@ func (bpp *basePreProcess) removeDataFromPools(
...
+		return process.ErrNilTxBlockBody
+	}
+	if check.IfNil(miniBlockPool) {
+		return process.ErrNilMiniBlockPool
+	}
+
+	for i := 0; i < len(body.MiniBlocks); i++ {
+		currentMiniBlock := body.MiniBlocks[i]
+		if !isMiniBlockCorrect(currentMiniBlock.Type) {
+			continue
```
[the remark this answers is not in the document]
Comment (SebastianMarian) [reply process/block/preprocess/basePreProcess.go:110]: Added

**Review by bogdan-rosianu — APPROVED** [review]

**Review by stefanandy — APPROVED** [review]
There was no match for bad tx in logs and the txs pools have been decreasing to almost nothing

**Pull request merged**